docs/plugins/ast.md

```diff
@@ -0,0 +1,117 @@
+# Generate MyST AST in plugins
+
+A common usecase with plugins involves generating MyST AST and inserting it into the document.
+This page covers a few ways that you can do so.
+
+## Parse MyST markdwon to AST in a directive or role
+
+The easiest way to generate MyST AST in a plugin is by using the `parseMyst` function in the `ctx` variable. It may be easier to parse MyST Markdown into AST nodes rather than [using the MyST sandbox](sandbox.md) to preview them.
+
+Here's an example of using the `parseMyst` function within a directive plugin to parse the _argument_ of the directive into MyST AST:
+
+```{code} javascript
+:filename: src/myplugin.mjs
+const myDirective = {
+  name: "mydirective",
+  doc: "My new directive!",
+  arg: { type: String, doc: "To be parsed into MyST AST." },
+  run(data, vfile, ctx) {
+    const ast = ctx.parseMyst(data.arg);
+    return ast.children[0];
+  },
+};
+```
+
+`ctx.parseMyst` returns a `root` node, which contains the parsed MyST.
+This function takes the first child of the root node, which is usually better for inserting into the document.
+
+You can also use this to manually construct MyST outputs using the arguments of your directive as data. For example:
```

Comment:
Still not 100% sure on encouraging use of `parseMyst` for constant nodes. @rowanc1 any thoughts on the implications of this?

Reply:
If we don't hear back, I will merge later today.

Reply:
What are your concerns with the workflow?

Alternatively we could remove the use of this from the tutorial (and revert to hand crafted AST), and out a big warning on the howto guide based on your concerns.